beps/0001-notifications-system/README.md

```diff
@@ -0,0 +1,255 @@
+
+In order to route signals from the sender to the intended receiver the signal plugin uses the concept of signaling channels. They are just like the topics on the message bus, but we use separate terminology in order to avoid confusion. They also sit one layer beneath the even topic in the protocol stack, where the signal channel is communicated as part of the event payload. In the frontend, the signal plugin exposes an API to subscribe to a signal channel. Each time the user receives a signal on the specified channel, a listener is called with the signal payload.
+
+### Notifications Plugin
+
+The role of the notifications plugin is to manage the lifecycle of notifications. The backend plugin provides an API for other backends to send notifications, as well as an accompanying [backend service](https://backstage.io/docs/backend-system/architecture/services). It also provides a separate API for the frontend plugin to read notifications for an individual user and manage the read status of notifications.
+
+The notification backend stores notification using the [database service](https://backstage.io/docs/backend-system/core-services/index#database). In particular it needs to store the following information for each notification:
+
+- ID
+- Payload
+- Receivers
+- Read status by user
+
+The receivers is **not** a list of users, but rather a filter that describes who should receive the notification. It must be possible to evaluate this filter in a database query, so that we can efficiently fetch all notifications for a given user. The same filter will also be used by the signal backend to determine which users should receive a signal.
+
+The notification backend does not provide any new permissions, since creating notifications can only be done by other backend plugins, while reading notifications can only be done by the authenticated user. It is possible that we want to add a permissions for reading notifications, in particular for admin and impersonation use cases, but that is not part of this proposal or the initial implementation.
+
+The notification frontend plugin provides a UI for viewing notifications, which in the initial implementation can be as simple as needed. The only requirement is that a user is able to view recent notifications and distinguish between read and unread notifications. A notification as marked as read once it has been interacted with. The frontend plugin also subscribes to the notifications signal channel and alerts the user when a new notification is received.
```

Comment:
The FE plugin should further provide FE API for other plugins to receive notifications.
Consuming FE plugins should be able to filter notifications they are interested in.

I will evolve this idea in the separate discussion issue.

Reply:
https://github.com/backstage/backstage/issues/22213#issuecomment-1889106559

Reply:
https://github.com/backstage/backstage/pull/22510

Reply:
Some updates are also in #21889. The directions this takes are still too different to get to the implementation phase. Would it make sense to have another meetup to go into details and figure out what to do?